test/test_datapipe.py

```diff
@@ -164,6 +176,7 @@ def tearDown(self):
         try:
             self.temp_sub_dir.cleanup()
             self.temp_dir.cleanup()
+            gc.collect()
```

Comment:
I tried adding this step before making changes, and it didn't resolve the issue related to unclosed buffer stream.

Reply:
This comment is just stating that add `gc.collect()` by itself doesn't resolve the issue. But the rest of the code changes will make the warnings disappear.

Reply:
If so, let's remove it because test should mimic normal user work flow that users would not call gc.collect() in most cases.

Reply:
Agree